sys/net/gnrc/transport_layer/tcp/gnrc_tcp.c

```diff
@@ -394,6 +395,23 @@ int gnrc_tcp_open(gnrc_tcp_tcb_t *tcb, const gnrc_tcp_ep_t *remote, uint16_t loc
             return -EINVAL;
         }
         tcb->ll_iface = remote->netif;
+
+        /* TCP connect does not work without specifying the network interface.
+         * First, try to find the right netif */
+        if (tcb->ll_iface == SOCK_ADDR_ANY_NETIF) {
+            if (gnrc_netif_highlander()) {
+                const gnrc_netif_t *netif = gnrc_netif_iter(NULL);
+                if (netif) {
+                    tcb->ll_iface = netif->pid;
+                }
+            }
+        }
+
+        /* We did not fine the netif, so rather than wasting more resources
+         * before timing out without sending anything, return an error */
+        if (tcb->ll_iface == SOCK_ADDR_ANY_NETIF) {
```

Comment:
```suggestion
        if (ipv6_addr_is_link_local(tcb->peer_addr) && tcb->ll_iface == SOCK_ADDR_ANY_NETIF) {
```

Reply:
I think for link local, we don't even need to check if only one netif is available: Any interface will be equally valid then, so we could always just take the first netif for link local addresses and unspecified netif.

Reply:
No, always chosing the first interface when no interface is specified is way too surprising/confusing.

Just return an error if you'd need to make a choice, like `gnrc_udp` does.

Reply:
I cannot find where UDP does this. Do you know where the code lives? If that is implemented already, I could split this out info a common function and reuse this. That way, we can ensure that TCP and UDP will behave consistently in this regard.

Reply:
OK, I just free coded this for now.